app/models/bank_account.rb

```diff
@@ -64,4 +62,18 @@ def set_initial_residue
   def to_s
     "#{name.truncate(30)} (#{Money::Currency.new(currency).symbol})"
   end
+
+  class << self
+    def grouped_by_currency(currency)
+      currencies = Currency.ordered(currency)
+
+      grouped = []
```

Comment:
@shir didn't understand how to do it via index_by. 
`BankAccount.all.index_by(:currency) ArgumentError: wrong number of arguments (1 for 0)
`
did it the next way:
```rb
  class << self
    def grouped_by_currency(def_currency)
      currencies = Currency.ordered(def_currency)

      all.group_by(&:currency).sort_by do |ba|
        currencies.index(ba.first)
      end
    end
  end
```